### ISSUE TITLE
простой wrapper для получения данных по тикеру

### ISSUE THREAD

**Issue by epogrebnyak** [WLM1ke/apimoex#8]
Можно идею опишу - для целей упрощения обучения (в вузе) и упрощения использования в целом?

У @WLM1ke - супербиблиотека, позволяющая вытащить, все что угодно, если знаешь, где лежит. Когда мы рассказываем что-то простое студентам - типа возьмите цену акции или облигации, скорее всего студенту / пользователю нужно:

- получить откуда-то список тикеров
- по этому тикеру из списка получить котировки

```python
tickers =  get_stock_tickers()
assert "AFLT" in 
a = Stock("AFLT")
df1 = a.get_quotes(start="2021-01-15")
df2 = a.get_quotes_ytd()
df3 = get_quotes_ytd(stock="AFLT")
```

Есть польза в такой обертке? Может API обертки должно быть какое-то другое?

**Comment by WLM1ke** [comment]
У меня была задача воспроизвести API биржи, как оно есть.

Я думаю, что ты глядишь сквозь призму своих задач. Кому-то нужны акции, а кому-то под какими тикерами торговалась бумага с таким-то регистрационным номером, кому-то нужны фьючерсы, кому-то нужно склеивать фьючерсы или акции с разными тикерами, а кому-то нет, кому-то дневные данные, кому-то минутные и т.д.

На самом деле задач может быть миллион, и не уверен, что конкретная обертка будет востребована. Если бы у проекта было 1000+ звезд и куча активных пользователей, то можно было бы понять, что некие требования реально существуют, тогда можно было бы их реализовать. 

Мне кажется практичнее в хлеп добавить страничку с некоторыми типовыми шагами использования или FAQ, как получить определенные данные.

**Comment by epogrebnyak** [comment]
Попалось на глаза: https://cran.rstudio.com/web/packages/moexer/readme/README.html

**Comment by WLM1ke** [comment]
Я не знаю, что там под капотом, но на первый взгляд это примерно мой вариант, а не продвинутая обертка, как ты предлагаешь:
search_security -  аналог apimoex.find_securities
get_security_info - аналог apimoex.find_security_description
get_candles - аналог apimoex.get_market_candles или apimoex.get_board_candles не знаю, как там реализовано внутри
get_candles - такого напрямую нет, но легко собрать из с помощью двух предыдущих вызовов

Можно написать подобный хелп, хотя судя по вопросам в issue в основном народ не это интересует. По памяти люди спрашивали как скачать фьючерсы, курс и котировки иностранных акций. В этом пакете не очень понятно как это сделать и можно ли сделать в принципе.

**Comment by epogrebnyak** [comment]
Да, сама идея helpa практически решает вопрос как использовать apimoeх для получения базовой инфы. 

Я попробуй структурировать какие данные показываются на сайте биржи (типа акции - TQBR, и т.д.).

Нашел код R: https://github.com/x1o/moexer/tree/master/R

**Comment by WLM1ke** [comment]
Жень - структурировать реально сложно. Сайт биржи много чего показывает. Мне кажется разумнее опираться на потребности пользователей. 

У тебя, как я понимаю, есть десятки, а то и сотни учеников, которым в рамках обучения будут использовать библиотеку в рамках неких стандартных заданий. Возможно разумно описать именно стандартный путь - например они выполняют задания только для российских акций и дневных котировок. Описать этот путь, а дальше можно при желании добавить несколько пунктов в виде FAQ - как скачать курс, как иностранные акции, как фьючерсы, если такие вопросы будут возникать, как скачать минутные данные и т.д.

**Comment by epogrebnyak** [comment]
@WLM1ke, а иерархия board/market/engine - она где-то внутри документации про ISS? Как в ней побыстрее разобраться?

```
board: str = "TQBR",
    market: str = "shares",
    engine: str = "stock",
```

**Comment by WLM1ke** [comment]
Документации полноценной нет - это некое профессионально знание, наверное оно разбросано по десяткам документов биржи и постоянно меняется. Так относительно недавно появились иностранные бумаги, иностранные бумаги в долларах, вечерняя и утренняя сессия и т.д. Вряд ли это все можно как-то отобразим и поддерживать в актуальном состоянии. 

Для разработчиков есть два документа, которые приведены на первой странице:
- Официальное руководство пользователя https://fs.moex.com/files/6523
- Перечень всех запросов https://iss.moex.com/iss/reference/

**Comment by WLM1ke** [comment]
И еще добавлю, эти документы тоже не полные - так реально есть возможность скачать дивиденды по некоторым бумагам, но это нигде не описано. Я случайно где-то подсмотрел, такой запрос - видимо человек раньше работал разработчики на бирже.

**Comment by epogrebnyak** [comment]
Спасибо, успокоил, что это разбросанное в разных доках знание и не стоит
его в одну стройную систему выстраивать. В целом я хочу сказать, что
Мосбиржа большие молодцы,что в принципе выдают эту информацию в свободном
API, без всяких регистраций, мне кажется уникальный случай.


пт, 19 нояб. 2021 г. в 08:11, WLMike ***@***.***>:

**Comment by WLM1ke** [comment]
Раньше видимо за деньги было, так как в документации разработчика описана авторизация, но когда я этим занялся, всю нужную мне информацию можно было получить без денег и регистраций. Сделано видимо давно, потому что сейчас модно описывать подобные сервисы с помощью OpenAPI/Swagger

**Comment by WLM1ke** [comment]
Тут вспомнил, что перечень всех существующих на данный момент board/market/engine  и т.д. можно получить с помощью запроса apimoex.get_reference или просто открыть в [браузере](https://iss.moex.com/iss/index.json) - их огромное количество. 

board/market/engine не свободно сочетаются между собой - надо знать как. И много из них сейчас не используются, но использовались ранее. Например сейчас основной режим торгов акций идет в режими T+2 или board TQBR, а раньше велся в режиме T, а до этого были отдельные режими для отдельных групп акций.

**Comment by epogrebnyak** [comment]
Классный JSON - большой, есть что поизучать! И securitygroups, и
securitycollections - ) хитрая логика как это все перекрещивается, но
здорово что в одном месте все лежит


API


пт, 19 нояб. 2021 г. в 11:13, WLMike ***@***.***>:

**Comment by epogrebnyak** [comment]
> Жень - структурировать реально сложно. Сайт биржи много чего показывает. Мне кажется разумнее опираться на потребности пользователей.

+++

**Comment by epogrebnyak** [comment]
Поразбирался как все работает тут: https://colab.research.google.com/drive/1gJZE2NPGH6_uJG1C3BPyPYvZyZQbRJts#scrollTo=r97xL2_cpIvU

(вроде есть функционал комментировать в ноутбуке также)

Хочу дальше облигации сделать

**Comment by WLM1ke** [comment]
> VALUE это не волюм на клоуз, а что-то другое

VALUE считается по цене конкретных сделок, а не по цене закрытия.

**Comment by WLM1ke** [comment]
> - replicate any stock index

Получится только на коротком интервале - состав и веса в индексе частенько меняют

**Comment by khpvo** [comment]
> И еще добавлю, эти документы тоже не полные - так реально есть возможность скачать дивиденды по некоторым бумагам, но это нигде не описано. Я случайно где-то подсмотрел такой запрос - видимо человек раньше работал разработчиком на бирже.

по сбербанку не все дивиденды показывает.

**Comment by DonRadja** [comment]
Камрады, а кто-то смог понять как получить с ISS MOEX данные по утренней сессии? Стандартные запросы типа http://iss.moex.com/iss/engines/stock/markets/shares/securities/YNDX/candles.json?from=2022-09-19&till=2022-09-19&interval=1&start=0
Возвращают только основную + вечернюю сессию. В документации ничего об этом нету. Прямо скажем, документация там не юзерфрендли чуть более, чем полностью (

**Comment by WLM1ke** [comment]
Утренние встречу встречал только в запросе истории котировок

https://iss.moex.com/iss/reference/813
https://iss.moex.com/iss/reference/817

**Comment by DonRadja** [comment]
Да, это я тоже нашел, а также  по фьючам, индексам. По фондовому рынку нету утренних данных в виде свечей.

Если вдруг кто голову ломает, то еще одна фишечка неочевидная к тому запросу, что я привел выше. Pagesize по умолчанию 500, параметр limit его не меняет.

**Comment by dani-kkk** [comment]
Добрый день. А есть способ получить цены для индекса IMOEX?

**Comment by WLM1ke** [comment]
get_market_history с параметром market="index" security="IMOEX"
